cpp/tests/join/distinct_join_tests.cpp

```diff
@@ -39,7 +39,17 @@ using strcol_wrapper = cudf::test::strings_column_wrapper;
 using CVector        = std::vector<std::unique_ptr<cudf::column>>;
 using Table          = cudf::table;
 
+std::unique_ptr<rmm::device_uvector<cudf::size_type>> get_left_indices(cudf::size_type size)
+{
+  auto sequence = std::vector<cudf::size_type>(size);
+  std::iota(sequence.begin(), sequence.end(), 0);
+  auto indices = cudf::detail::make_device_uvector_sync(
+    sequence, cudf::get_default_stream(), rmm::mr::get_current_device_resource());
+  return std::make_unique<rmm::device_uvector<cudf::size_type>>(std::move(indices));
```

Comment:
Would it be better to use `cudf::sequence` here and construct this on the device? Then maybe release it into an `rmm::device_uvector` if needed.

Reply:
Yeah, I've thought about this. My blocking point was that `cudf::sequence` returns a unique pointer of an integer column. Unlike `device_vector`, `device_uvector` cannot be constructed from a pointer with a size (or a span) so having the column doesn't really help.

Reply:
349d853

Reply:
I'm fine with having this in the test code. I believe that we can still construct `device_uvector` from the output of `cudf::sequence`, but that will need some sort of non-trivial (if not complex) work to convert the content of `column::release()` (which is `unique_ptr<rmm::device_buffer>`) into a `unique_ptr<device_uvector>`.